packages/playwright-test/src/transform.ts

```diff
@@ -100,23 +100,52 @@ export function resolveHook(filename: string, specifier: string): string | undef
     return;
   const tsconfig = loadAndValidateTsconfigForFile(filename);
   if (tsconfig) {
+    let longestPrefixLength = -1;
+    let pathMatchedByLongestPrefix: string | undefined;
+
     for (const { key, values } of tsconfig.paths) {
-      const keyHasStar = key[key.length - 1] === '*';
-      const matches = specifier.startsWith(keyHasStar ? key.substring(0, key.length - 1) : key);
-      if (!matches)
-        continue;
+      let matchedPartOfSpecifier = specifier;
+
+      const [keyPrefix, keySuffix] = key.split('*');
+      if (key.includes('*')) {
+        // * If pattern contains '*' then to match pattern "<prefix>*<suffix>" module name must start with the <prefix> and end with <suffix>.
+        // * <MatchedStar> denotes part of the module name between <prefix> and <suffix>.
+        // * If module name can be matches with multiple patterns then pattern with the longest prefix will be picked.
+        // https://github.com/microsoft/TypeScript/blob/f82d0cb3299c04093e3835bc7e29f5b40475f586/src/compiler/moduleNameResolver.ts#L1049
+        if (keyPrefix) {
+          if (!specifier.startsWith(keyPrefix))
+            continue;
+          matchedPartOfSpecifier = matchedPartOfSpecifier.replace(keyPrefix, '');
```

Comment:
Actually this `replace` can also be replaced with `substring` for the sake of speed, let me do it as well